client/data/emails/use-get-email-accounts-query.ts

```diff
@@ -1,7 +1,13 @@
 import { useQuery } from 'react-query';
 import wpcom from 'calypso/lib/wp';
 
-export const getCacheKey = ( siteId, domain ) => [ 'emailAccounts', siteId, domain ];
+type UseQueryParams = Parameters< typeof useQuery >;
+
+export const getCacheKey = ( siteId: number | null, domain: string ) => [
```

Comment:
It shouldn't be, but Typescript interprets the JSDoc type definition in [calypso/state/ui/selectors/get-selected-site-id.js](https://github.com/Automattic/wp-calypso/blob/trunk/client/state/ui/selectors/get-selected-site-id.js) as `number | null` and we need to adapt to that.

In a broader perspective, I suppose this makes sense, since `selectedSiteId` could be null at other times in Calypso.